.github/workflows/build-and-publish.yml

```diff
@@ -44,3 +47,17 @@ jobs:
           nexus_username: ${{ secrets.MAVEN_USERNAME }}
           nexus_password: ${{ secrets.MAVEN_PASSWORD }}
           server_id: ossrh
+
+      - name: Create GitHub release
```

Comment:
Prior art from pay-telegraf https://github.com/alphagov/pay-telegraf/blob/main/.github/workflows/tag-on-merge-to-main.yml#L35